modules/core/include/opencv2/core/utility.hpp

```diff
@@ -513,30 +513,36 @@ class CV_EXPORTS AutoLock
     AutoLock& operator = (const AutoLock&);
 };
 
+// TLS interface
 class CV_EXPORTS TLSDataContainer
 {
-private:
-    int key_;
 protected:
-    TLSDataContainer();
-    virtual ~TLSDataContainer();
-public:
-    virtual void* createDataInstance() const = 0;
-    virtual void deleteDataInstance(void* data) const = 0;
+    TLSDataContainer() { key = (size_t)-1; };
+    virtual ~TLSDataContainer() {};
 
+    void  getKey();
     void* getData() const;
+    void  releaseKey();
+
+private:
```

Comment:
It is not supported by build system and we haven't yet git branch for 4.x development (There is not plans to add it before 3.1 release).
So this code block should not break current build and should looks like a TODO for future changes (You can also put a TODO here instead).

ABI changes are not welcome in general. But it can be approved for experimental modules, features. Almost all experimental modules are in opencv_contrib at this moment.

P.S. There is a mistake in my `#if` condition above. It should not break current build.